meteo/_hycom.py

```diff
@@ -0,0 +1,168 @@
+    day: Annotated[L_Days, Parameter(show_choices=False)],
+    dataset: L_HYCOM_Dataset = "Operational",
+    output_dir: Path | None = None,
+):
+    """
+    Parameters
+    ----------
+    year : int
+        Year to download (must be >= 1900).
+    month : int
+        Month to download (1-12).
+    day : int
+        Day to download (1-31).
+    output_dir : Path, default: system-specific
+        Output directory for downloaded files. Data is saved to a 'HYCOM'
+        subdirectory within this path. Defaults to a platform-specific location.
+    dataset: str
+        If the dataset type is "Operational" or "Reanalysis"
+    """
+
+    if dataset == "Reanalysis":
+        raise NotImplementedError("Reanalysis not implemented yet")
+
+    logger.debug("Saving to: %s", output_dir)
+    output_dir.mkdir(parents=True, exist_ok=True)
+
+    bbox = Bbox()
+    date = pd.Timestamp(year, month, day)
+
+    database = get_database(date)
+    logger.info(f"Fetching data for {date} from database {database}")
+
+    time_idx, lon_idx1, lon_idx2, lat_idx1, lat_idx2 = get_idxs(date, database, bbox)
+    selection_str = f"[{lat_idx1}:{SUB_SAMPLE}:{lat_idx2}][{lon_idx1}:{SUB_SAMPLE}:{lon_idx2}]"
+
+    url_ssh = (
+        f"https://tds.hycom.org/thredds/dodsC/{database}?lat[{lat_idx1}:{SUB_SAMPLE}:{lat_idx2}],"
+        + f"lon[{lon_idx1}:{SUB_SAMPLE}:{lon_idx2}],depth[0:1:-1],time[{time_idx}],"
+        + f"surf_el[{time_idx}]{selection_str},"
+        + f"water_u[{time_idx}][0:1:{HYCOM_MAX_DEPTH_INDEX}]{selection_str},"
+        + f"water_v[{time_idx}][0:1:{HYCOM_MAX_DEPTH_INDEX}]{selection_str},"
+        + f"water_temp[{time_idx}][0:1:{HYCOM_MAX_DEPTH_INDEX}]{selection_str},"
+        + f"salinity[{time_idx}][0:1:{HYCOM_MAX_DEPTH_INDEX}]{selection_str}"
+    )
+
+    foutname = f'{output_dir}/hycom_{date.strftime("%Y%m%d")}.nc'
+    logger.info(f"filename is {foutname}")
+    ds = xr.open_dataset(url_ssh)
+
+    # convert in-situ temperature to potential temperature
+    temp = ds.water_temp.values
+    salt = ds.salinity.values
+    dep = ds.depth.values
+
+    ptemp = ConvertTemp(salt, temp, dep)
+    # drop water_temp variable and add new temperature variable
+    ds = ds.drop("water_temp")
+    ds["temperature"] = (["time", "depth", "lat", "lon"], ptemp)
+    ds.temperature.attrs = {
+        "long_name": "Sea water potential temperature",
+        "standard_name": "sea_water_potential_temperature",
+        "units": "degC",
+    }
+
+    ds = ds.rename_dims({"lon": "xlon"})
+    ds = ds.rename_dims({"lat": "ylat"})
+    ds = ds.rename_vars({"lat": "ylat"})
+    ds = ds.rename_vars({"lon": "xlon"})
+
+    t0 = time()
+    logger.info("Start writing nc file!")
+    ds.to_netcdf(
+        foutname,
+        "w",
+        unlimited_dims="time",
+        encoding={
+            "temperature": {
+                "dtype": "h",
+                "_FillValue": -30000.0,
+                "scale_factor": 0.001,
+                "add_offset": 20.0,
```

Comment:
Using a scale_factor and an offset without compression does not make much sense. If we want to use them let's calculate them properly and add compression, too. if not, let's not remove them.